### ISSUE TITLE
App Crash in settings

### ISSUE THREAD

**Issue by Ahmed-Abd-Elsamie** [amahi/android#529]
The app crashes when changing between light and dark theme

![Screenshot_2020-03-30-20-54-05-484_com miui home](https://user-images.githubusercontent.com/41450463/77959652-3c189300-72d7-11ea-99cf-ae5df640362c.jpg)

**Comment by chirag-jn** [comment]
Is this an issue in every phone or is it specific to the manufacturer of your device?

**Comment by Ahmed-Abd-Elsamie** [comment]
Yes, i tried running the app on multiple devices and it's still crashing

**Comment by chirag-jn** [comment]
@Ahmed199810 Seems weird. Can you please give the specifications of those devices?

**Comment by Ahmed-Abd-Elsamie** [comment]
The first is real device (Xiaomi Redmi note 7 Android 9)
The second is emulator Nox with Android 5
@chirag-jn

**Comment by chirag-jn** [comment]
@Ahmed199810 could you figure out what went wrong?

**Comment by Ahmed-Abd-Elsamie** [comment]
@chirag-jn I have figured out the problem and fixed it.
the problem was in the NavigationActivity.class , after changing between light and dark theme we finish the activity then start it again from itself. and this will cause the app to crash because we are trying to start it from null context.

Solution :
i removed startActivityForResult function and replaced it with the following:
1 - once opening the settings page i finish the navigationActivity.
2 - once do any change in the settings like changing between light and dark themes, we start the NavigationActivity again from the settingActivity.

I tried this method and solved the problem

here is the the code changes  
https://github.com/amahi/android/pull/560/commits/775486d6a8663296afe9f8206b2c3edd517eecd3
